### PR TITLE
fix(card, combobox-item): rerender components with conditional slots when slotting elements after initialization

### PR DESCRIPTION
**Pull request by driskull**
**Related Issue:** #3686

## Summary

fix(card, combobox-item): Update to use ConditionalSlotComponent

### CODE REVIEW

File: src/components/calcite-combobox-item/calcite-combobox-item.e2e.ts [hunk 1]
```diff
@@ -0,0 +1,10 @@
+import { hidden, renders } from "../../tests/commonTests";
+//import { SLOTS } from "./resources";
+
+describe("calcite-combobox-item", () => {
+  it("renders", async () => renders("calcite-combobox-item", { display: "flex" }));
+
+  it("honors hidden attribute", async () => hidden("calcite-combobox-item"));
+
+  //it("has slots", () => slots("calcite-combobox-item", SLOTS));
```
Comment (driskull) [inline src/components/calcite-combobox-item/calcite-combobox-item.e2e.ts:9]: @jcfranco I think we need to enhance the `slots` commonTest to be able to test the default slot. Can we do that in this PR?

Comment (jcfranco) [reply src/components/calcite-combobox-item/calcite-combobox-item.e2e.ts:9]: I'll take care of that today.

Comment (jcfranco) [reply src/components/calcite-combobox-item/calcite-combobox-item.e2e.ts]: Helper updated and test wired up. 🎉

**Review by jcfranco — APPROVED** [review]
Thanks for splitting this up! 💪🏼 

FYI, I updated the PR title to match the previous PR with similar fixes.

**Pull request merged**